analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -478,27 +477,52 @@ public void Branching_LearnsObjectConstraint_ConstantPattern_Null_Negated(string
         public void Branching_LearnsObjectConstraint_ConstantPattern_True(string expression, string argType = "object")
         {
             var validator = CreateIfElseEndValidatorCS(expression, OperationKind.ConstantPattern, argType);
-            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(BoolConstraint.True));
-            validator.ValidateTag("Else", x => x.Should().BeNull("it could be False, null or any other type"));
-            validator.TagValues("End").Should().HaveCount(2)
-                .And.ContainSingle(x => x == null)
-                .And.ContainSingle(x => x != null && x.HasConstraint(BoolConstraint.True));
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(BoolConstraint.True)); // Should be NotNull, True
+            validator.ValidateTag("Else", x => x.Should().HaveNoConstraints("it could be False, null or any other type"));
+            validator.TagValues("End").Should().SatisfyRespectively(
+                x => x.Should().HaveOnlyConstraints(BoolConstraint.True),
```

Comment:
`true` without `NotNull`? So there's one more case somewhere to solve? Can you add to track on the parent issue?